File: backend/app/domain/repositories/feeding_repository.py
from abc import ABC, abstractmethod
from datetime import date
from uuid import UUID

from app.domain.entities.feeding import Feeding


class FeedingRepository(ABC):
    @abstractmethod
    async def get(self, id: UUID) -> Feeding | None:
        ...

    @abstractmethod
    async def get_by_baby_and_date(self, baby_id: UUID, target_date: date) -> list[Feeding]:
        ...

    @abstractmethod
    async def get_recent(self, baby_id: UUID, limit: int = 12) -> list[Feeding]:
        ...

    @abstractmethod
    async def save(self, feeding: Feeding) -> Feeding:
        ...

    @abstractmethod
    async def update(self, feeding: Feeding) -> Feeding:
        ...

    @abstractmethod
    async def delete(self, id: UUID) -> None:
        ...

File: zzippu_ios/zzippu/Domain/Repositories/FeedingRepository.swift
// Domain/Repositories/FeedingRepository.swift
// Foundation only — SwiftUI/SwiftData import 금지

import Foundation

protocol FeedingRepository {
    func create(_ feeding: Feeding) async throws -> Feeding
    func update(_ feeding: Feeding) async throws -> Feeding
    func delete(id: UUID, babyId: UUID) async throws          // 서버 경로에 babyId 필요, 물리삭제(204)
    func fetch(id: UUID, babyId: UUID) async throws -> Feeding?
    func list(babyId: UUID, on day: Date) async throws -> [Feeding]
    func lastFeeding(babyId: UUID) async throws -> Feeding?   // list().first로 구현

    // MARK: - Range API (달력 월별 집계용, S3 추가)
    // 기존 list(on:)는 유지(하위호환). range는 추가만.
    /// KST 날짜 범위에 속하는 날별 총 수유량 집계 반환.
    /// - Parameters:
    ///   - babyId: 아기 UUID
    ///   - start: 범위 시작 KST 자정 (포함)
    ///   - end:   범위 끝 KST 자정 (포함)
    /// - Returns: [DateVolume] — day = KST 자정, totalMl = 당일 합산
    func dailyTotals(babyId: UUID, from start: Date, to end: Date) async throws -> [DateVolume]
}
